### PR TITLE
Mpbn support

### PR DESCRIPTION
**Pull request by kyuhyongpark**
#50 
Created new condition MPBN_update for many functions

In case of MPBN_update:
Time reverse stable motifs are not calculated
new function finding source motifs
new check for terminal reduction
function creating attractor dictionary is modified

### CODE REVIEW

File: Testing/BasicFunctionsExample.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import PyBoolNet
```
Comment (jcrozum) [inline Testing/BasicFunctionsExample.py:1]: Could you move the MPBN demonstration to its own file and revert the changes here? I'm not 100% sure, but I think we reference the contents of this file in some documentation.

File: PyStableMotifs/AttractorRepertoire.py [hunk 2]
```diff
@@ -60,7 +60,7 @@ def __init__(self):
         self.relevant_nodes = None
 
     @classmethod
-    def from_primes(cls,primes,max_simulate_size=20,max_stable_motifs=10000):
+    def from_primes(cls,primes,max_simulate_size=20,max_stable_motifs=10000,MPBN_update=False):
```
Comment (jcrozum) [inline PyStableMotifs/AttractorRepertoire.py:63]: Because new functions and arguments have been added, we need to update the docs before merging. Do you know how to run `Build_Manual.bat` (on Windows) or `Build_Manual.sh` (on Mac/Linux)? You will need to install `pandoc` and `XeLaTeX`. We can talk more about how to set this via email.

File: PyStableMotifs/Reduction.py [hunk 3]
```diff
@@ -514,6 +530,59 @@ def merge_source_motifs(self):
...
+        source nodes all stabilize simultaneously.
+
+        Assumes that stable_motifs have already been computed,
+        but time_reverse_primes and time_reverse_stable_motifs are not.
+
+        To be used in the case of MBPN update.
+
+        Parameters
+        ----------
+        primes : dictionary of lists of lists of dictionaries
```
Comment (jcrozum) [inline PyStableMotifs/Reduction.py:546]: Throughout the documentation, we refer to this data type as a "PyBoolNet primes dictionary", and you don't need to give an example here. So this should be something like this
```python
"""Merges stable motifs . . . 
Parameters
----------
primes : PyBoolNet primes dictionary
     PyBoolNet update rules whose source node stable motifs are to be merged.

Returns
. . .
"""
```

File: PyStableMotifs/Reduction.py [hunk 4]
```diff
@@ -935,6 +1004,43 @@ def generate_attr_dict(self):
                 # attr_list.append(node_state_dict)
             return attr_list
 
+    def simple_generate_attr_dict(self, MPBN_update=False):
```
Comment (jcrozum) [inline PyStableMotifs/Reduction.py:1007]: Can we merge this function into `generate_attr_dict` instead (which would take `MPBN_update` as an argument)? The MPBN_update argument and assertion in this function is a little unusual. While we're at it, we might want to rename `generate_attr_dict` to `generate_attr_dict_list`.

**Review by jcrozum — CHANGES_REQUESTED** [review]
The code looks good and seems to work. There are some minor documentation and organization things that need to be addressed before we can merge and release.

File: PyStableMotifs/AttractorRepertoire.py (hunk 2 above)
Comment (jcrozum) [reply PyStableMotifs/AttractorRepertoire.py:63]: Oh, and one more thing I forgot to mention: we should add the MPBN reference to the documentation here. Here's the ref.:
L Paulevé, J Kolčák, T Chatain, S Haar, "Reconciling qualitative, abstract, and scalable modeling of biological networks." Nat. Com. vol. 11, no. 4256 (2020)

File: Testing/BasicFunctionsExample.py (hunk 1 above)
Comment (troonmel) [reply Testing/BasicFunctionsExample.py]: I left the original BasicFunctionsExample.py untouched in the Example and Tutorials folder.
I renamed the copied file in Testing folder to MPBN_BasicFunctionsTesting

File: PyStableMotifs/Reduction.py (hunk 3 above)
Comment (troonmel) [reply PyStableMotifs/Reduction.py]: I fixed it.
It was more like a note for myself.

File: PyStableMotifs/Reduction.py (hunk 4 above)
Comment (troonmel) [reply PyStableMotifs/Reduction.py]: I merged them and changed the name to generate_attr_dict_list.
I also fixed the name where the function is called

**Comment by troonmel** [comment]
I tried MPBN update for all codes in Example and Tutorials Folder.

TLGL_Small_analysis.py gives an error because it tries to print time reversal stable motifs,
but no other errors than that.

File: PyStableMotifs/AttractorRepertoire.py (hunk 2 above)
Comment (troonmel) [reply PyStableMotifs/AttractorRepertoire.py:63]: Where should I add the MPBN reference? I tried adding it whenever there's a MPBN_update parameter for a function.

**Review by jcrozum — APPROVED** [review]
